legend-engine-xts-graphQL/legend-engine-xt-graphQL-query/src/main/java/org/finos/legend/engine/query/graphQL/api/execute/GraphQLExecute.java

```diff
@@ -342,6 +374,30 @@ private List<SerializedNamedPlans> buildPlanWithParameter(String queryClassPath,
         return plans;
     }
 
+    private Map<String, ?> computeExtensionsField(OperationDefinition operationDefinition)
+    {
+        List<Directive> directives = ((Field)(operationDefinition.selectionSet.get(0))).directives.stream().distinct().collect(Collectors.toList());
+        if (directives.isEmpty())
+        {
+            return new HashMap<>();
+        }
+        String fieldName = ((Field)(operationDefinition.selectionSet.get(0))).name;
+        Map<String, Map<String,Object>> m = new HashMap<>();
+        m.put(fieldName, new HashMap<>());
+        directives.forEach(directive ->
+        {
+            if (directive.name.equals("echo"))
```

Comment:
Noted, I will design it for extension in next changeset, while introducing a new directive called `totalCount`. It would make use of elements such as execution plans, plan executor which will help to reason about the mechanism.